src/utils.js

```diff
@@ -0,0 +1,11 @@
+export function urlFormat(url, {filename=null, version=null}={}) {
+  if (!filename || !version) {
+    throw new Error('ArgumentError: File and version are required.');
```

Comment:
This is likely a limitation in JavaScript but can we have a more specific error type? Storing that information in the Error message and regexing for it seems like it could cause pain down the line.